temporalio/bridge/src/lib.rs

```diff
@@ -0,0 +1,296 @@
+use pyo3::exceptions::{PyRuntimeError, PyValueError};
+use pyo3::prelude::*;
+use std::collections::HashMap;
+use std::time::Duration;
+use temporal_client::WorkflowService;
+use tonic;
+
+#[pymodule]
+fn temporal_sdk_bridge(_py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_class::<ClientRef>()?;
+    m.add_function(wrap_pyfunction!(new_client, m)?)?;
+    Ok(())
+}
+
+type Client = temporal_client::RetryGateway<
+    temporal_client::ConfiguredClient<temporal_client::WorkflowServiceClientWithMetrics>,
+>;
+
+#[pyclass]
+pub struct ClientRef {
+    retry_client: Client,
+}
+
+#[derive(FromPyObject)]
+pub struct ClientOptions {
```

Comment:
I think this is the most convenient. Making classes in Rust that are created and set in Python is more fickle.

The duplicate definitions have to exist regardless because Rust does not auto generate types for Python.

The same thing basically happens in TypeScript but even worse. You have a definition in `core-bridge/index.d.ts` and then untyped, unchecked, string-based definition assumption inside Rust during conversion. This at least lets you add types to both sides. We _could_ take a raw dictionary and lose all checking, but you still effectively have two places where you define the structure - lang and Rust.

Reply:
SGTM